scala/scala_import.bzl

```diff
@@ -20,27 +21,45 @@ def _scala_import_impl(ctx):
         ),
         jars_to_labels = jars2labels,
         providers = [
-            _create_provider(current_jars, transitive_runtime_jars, jars, exports)
+            java_common.merge([jars_provider, deps_provider, runtime_deps_provider, exports_provider])
         ],
     )
-def _create_provider(current_target_compile_jars, transitive_runtime_jars, jars, exports):
-    return java_common.create_provider(
-        use_ijar = False,
-        compile_time_jars = depset(transitive = [current_target_compile_jars, exports.compile_jars]),
-        transitive_compile_time_jars = depset(transitive = [jars.transitive_compile_jars, current_target_compile_jars, exports.transitive_compile_jars]) ,
-        transitive_runtime_jars = depset(transitive = [transitive_runtime_jars, jars.transitive_runtime_jars, current_target_compile_jars, exports.transitive_runtime_jars]) ,
-      )
+
+def _jars_to_provider(jars):
+  providers = []
+  for jar in jars:
+    if JavaInfo in jar:
+      fail("jars must contain only jar files")
```

Comment:
should we have a test for this negative case?

Reply:
and also the error message is about what it should contain but the check is about whether or not this has JavaInfo.
Isn't there a mix? Maybe have a failure here to say targets of jvm_rules aren't allowed here and add another check below to see that the files are only `jar` files